lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -17,6 +18,14 @@ def get_identity(self, data):
         except AttributeError:
             return None
 
+    def validate(self, data):
+        data = super(CourseModeSerializer, self).validate(data)
+        if 'professional' in data['mode_slug'] and data['expiration_datetime'] is not None:
```

Comment:
Discussed offline, and I'm fine with this to address the immediate bug.

Only suggestion I'd make here is to delegate the check for whether the mode is professional to the course modes app.  I think there's a `CourseMode.is_professional_mode()` that you can use or adapt, just in case we later decide to add other professional modes (or change the mechanism by which modes are marked as being professional).